docs/docs/alephium/sdk.md

```diff
@@ -0,0 +1,656 @@
+)
+
+// print transaction id
+console.log(createPositionTransactionId)
+
+// check the newly opened position
+console.log(await invariant.getPosition(account.address, 0n))
+```
+
+:::tip Output
+Token X amount:  7999999999880n  Token Y amount:  8000000000000n <br/>
+69226b27395a6820e99c761566439106d04f02dec3833301b25c5531262368d7<br/>
+{<br/>
+&emsp;  poolKey: {<br/>
+&emsp; &emsp;   tokenX: 'a9ec8420ab99aa433645da0a0462ebe07351f0e6cdd56e1f3149d68dc6783300',<br/>
+&emsp; &emsp;   tokenY: 'e697a8bc5ea2433eaa8c0ce05f2730f5a17d654e93dbb459f114c88359d3d800',<br/>
+&emsp; &emsp;   feeTier: { fee: [Object], tickSpacing: 1n }<br/>
+&emsp;  },<br/>
+&emsp;  liquidity: 1600480031975990558848n,<br/>
+&emsp;  lowerTickIndex: -10n,<br/>
+&emsp;  upperTickIndex: 10n,<br/>
+&emsp;  feeGrowthInsideX: 0n,<br/>
+&emsp;  feeGrowthInsideY: 0n,<br/>
+&emsp;  lastBlockNumber: 1723210526545n,<br/>
+&emsp;  tokensOwedX: 0n,<br/>
+&emsp;  tokensOwedY: 0n,<br/>
+&emsp;  owner: '13FgGApAthBNZGwrUiPApixwxaNdfSxvidNfbPPLdAbRm',<br/>
+&emsp;  exists: true<br/>
+}<br/>
+:::
+
+### Swap tokens
+
+Performing a swap requires: specifying the amount of tokens to be swapped or desired amount to receive from the swap (input token amount will be calculated durning the swap), approving the transfer of the token, estimating the result of the swap, direction, determining the allowed slippage, calculating the square root price limit based on slippage, and finally, executing the swap. It's essential to note that the swap tolerance is expressed in square root price (sqrtPrice) after the swap, rather than the amount of tokens.
+
+:::note Price impact and slippage
+Price impact represents the change in price observed after the completion of a swap. It provides insight into how the executed swap influences the token price within the liquidity pool. A higher price impact indicates a more significant alteration in the token price post-swap.
+
+Slippage refers to the difference between the estimated square root of price after a swap is initiated and the actual square root of price observed after the swap is executed. It quantifies the deviation between the expected and realized square roots of prices. Slippage does not imply an acceptable threshold solely in terms of token amounts.
+
+While price impact focuses on the post-swap change in token price within the liquidity pool, slippage highlights the variance between the anticipated and actual prices during and after the swap process.
+:::
+
+```typescript
+// we want to swap 6 token0
+// token0 has 12 decimal places
+const amount = toTokenAmount(6n, 12n)
+
+// get estimated result of swap - there are 2 ways to do it
+// 1. use the quote method
+// due to it being computed using blockchain, thus having a latency and being subjected to gas limit, we recommend the second method
+const quoteResult = await invariant.quote(poolKey, true, amount, true, await getMinSqrtPrice(feeTier.tickSpacing))
+
+// 2. use local simulation of a swap [PREFERRED]
```

Comment:
I dont think we should suggest the simulation.

Reply:
let's leave a suggestion to use swap simulation